### DISCUSSION TITLE
RetroNAS on a QNAP TS-563 - CIFS or NFS?

### DISCUSSION THREAD

**Discussion by iamaburrama** [retronas/discussions#121] [Q&A] [answered]
Hello all,

I'm trying to install RetroNAS on my QNAP TS-563 using Debian 11.2 on Virtualization Station.

I've been having varying degrees of success with each method while at the same time hitting walls getting things to run properly.

To try to save my sanity I was wondering if there was a recommended protocol to mount storage?  Or if anyone has any experience using a QNAP for this application?

For each approach I restore the nas to factory defaults just to make sure I don't have any lingering settings.

If I mount using SMB/CIFS through fstab all works well enough.  I'm able to view the share, copy files from my previous retronas install, etc. but when I install the samba mister directories it seems to create files that contain symlinks in areas where directories should be.  I'm not sure if this is an issue with the qnap or possibly permissions? 

If I mount using NFS through fstab I seem to get slightly better results.  If I create samba mister directories all folders seem to be created however if I try to import files from my previous server install I don't have permissions to write to any of the created folders.  Prior to the install, on the directory I use for storage on the NAS side I'm able to create folders/files via a networked windows pc using windows explorer as well as create folders via ssh when RetroNAS is mapped.  But as soon as I install directories I get an access denied message.  

It sounds like NFS may have some protocol limitations so I believe SMB would be more recommended? 

Any guidance on what way I should approach this would be greatly appreciated.

**Comment by sairuk** [comment]
NFS should be fine, we don't recommend anything atm in regards to hosting on a NAS although I know folks have had some successes. Based on others reports I'd assume your problems with NFS are potentially down to the acl mapping between the VM and the host NAS. 

I have a older QNAP and a Terramaster and found both of them fall short in the ui nfs implementations. QNAP is much better however than TM. My QNAP is unfortunately too old to be hosting VMs otherwise i'd have documented an approach.

You will miss out on some functionality and NFS cannot be reshared atm from the VM if it is an NFS mount.

Hopefully the community can chime in.

**Reply by iamaburrama** [reply]
So I've been playing around with the QNAP and it looks like I can mount using iscsi.  

Total noob but looking into this it seems like it might potentially behave as well as using usb for storage?  (I'm sure there's much more nuance to it than that.)  I also was thinking this might be a better workaround to the NFS issues I think you were alluding to in your previous post (I also would love to use ps3netsrv if possible.) 

Either way, I was able to create an iscsi lun on the qnap ui.  

Then went through your VM walkthrough install video.

Once completing these steps I followed this [qnap forum post ](https://forum.qnap.com/viewtopic.php?p=753521&hilit=mount#p753521) to mount the drive in debian vm in virtualization station.

 
After mount I used your directions in the Installing Retronas Wiki article.

Installed cockpit, then went over to the web interface, formatted, mounted and named the iscsi target. (In this case, /storage)

To test things out I've tried to install the mister cifs shares just to create directories and establish the \\retrosmb\ share.

I try to pull up the retronas share on a windows pc, when I connect to it I get prompted for a password (I'm using generics from your instructions, user and password pi standard uid/gid) this opens up the mister and retronas share.  If I try to open either folder I get prompted again for credentials - but this time it seems pi/pi doesn't work.

I feel like I'm overlooking something at this point.  I've tried to look through discussions and the subreddits but haven't found anything near the same problem.

Is this something silly?

**Reply by sairuk** [reply, marked as answer]
you will need to set the samba credentials through our config menu after installing a samba service before attempting to access the shares, make sure you have done this

as to how windows works with storing samba credentials for shares, no idea sorry I haven't used windows as an OS for 20y.

**Reply by iamaburrama** [reply]
I swear I tried this a few times and it still didn't work.  

I also looked over your samba/cifs video to make sure I was following your previous directions accurately.  

I walked away from it for a few hours, came back, and all of a sudden it worked!

Really appreciate your help!  Hoping this config info will be helpful to others.

**Comment by iamaburrama** [comment]
One more thing - I'm copying over my media from my old server's external usb hard drive.  

I have the drive mounted and was wondering what the best way would be to copy everything over.

I'm assuming I'd want to copy over the roms, saves, savestates and wallpapers from the old drive?  I'm thinking the rest of the other folders are symlinks..? (in my case - ps2, ps3, gamecube)

Is the standard cp commands best here or is there a way to watch progress in ssh?  Was seeing navigator as a possible option but thinking I might overload it with such a large file transfer.

**Reply by sairuk** [reply]
Probably `rsync` .... something like

`rsync -avzP <source> <dest>`

to test/preview the changes its going to make do a `--dry-run` and make sure you are happy with the changes.

`rsync -avzP --dry-run <source> <dest>`

you can read about the options with `man rsync` in the terminal
